src/Juvix/Prelude/Base.hs

```diff
@@ -589,11 +590,14 @@ runInputInfinite s =
             return i
       )
 
-writeFileEnsureLn :: (MonadMask m, MonadIO m) => FilePath -> Text -> m ()
-writeFileEnsureLn p t =
-  let t' = case Text.unsnoc t of
-        Nothing -> t
-        Just (_, y) -> case y of
-          '\n' -> t
-          _ -> Text.snoc t '\n'
-   in writeFile p t'
+ensureLn :: Text -> Text
+ensureLn t =
+  case Text.unsnoc t of
+    Nothing -> t
```

Comment:
Something I didn't think about at the time - should we add a newline if the text is empty?

Reply:
I don't think so. See [definition of Text File](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_403).

> A file that contains characters organized into zero or more lines. [..]

It is allowed for a text file to have no lines, then the newline character should not be added.

Reply:
That's right! Thanks